### DISCUSSION TITLE
Branch VS Fork

### DISCUSSION THREAD

**Discussion by WaleedMortaja** [community/community#22846] [New to GitHub]
<p>If I work with a team on a project on GitHub. And I want to make edits to the project. Should I make a new <em>branch</em> for the edition or make a new <em>fork</em> ?</p>
<p>Please note that the team has right access on the repository so any member have to option to branch/fork.</p>
<ul>
<li>Branch supporters (from our team):
<ul>
<li>From GitHub Learning Lab course <a href="https://lab.github.com/courses/managing-merge-conflicts" rel="nofollow noopener">managing-merge-conflicts</a> we understood that edits comes as branches and then make a pull requests to the master (or to primary branch if it is not master) and then delete the branch after merging (or close and stop working on it).</li>
<li>And this link says the same: <a href="https://help.github.com/articles/github-glossary/#branch" rel="nofollow noopener">github-glossary/#branch</a>
</li>
<li>This page too <a href="https://help.github.com/articles/about-branches/" rel="nofollow noopener">about-branches</a> says that:</li>
</ul>
</li>
</ul>
<blockquote>
<p>You can use branches to:</p>
<ul>
<li>Develop features</li>
<li>Fix bugs</li>
<li>Safely experiment with new ideas</li>
</ul>
</blockquote>
<ul>
<li>Some video about GitHub (I think they were from GitHub itself) said that you use branches for edits then merge then delete (I don’t have the link for the video)</li>
<li>Fork supporters (from our team):
<ul>
<li>Users with right access to the repository, who can make branches, can also edit the branches of the other team members without the other memmber’s consent , while making a fork deny such an edition withiout permission of fork owner (even if we, in our team, don’t mess our each others work, it stills available).</li>
<li>Alot of known companies on GitHub use brnaches to provide the code for different versions of the project, while using forks for editing.</li>
</ul>
</li>
</ul>
<p>So should we use branches or forks for editing the project ?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>There’s no right or wrong way to do it if it achieves your goals. But, the most common ways people use branches and forks for collaboration is:</p>
<ul>
<li>People who are members of the organization use branches in the organization’s repository</li>
<li>People who are not members of the organization use branches in forks of the organization’s repository</li>
</ul>
<p>So, for example, I am a member of the <a href="https://github.com/atom" rel="nofollow noopener">Atom organization</a>. When I want to suggest a change to anything in the Atom project, I create a branch locally, work on my changes, then push the branch to the GitHub-hosted repo, and open a pull request. However, I am not a member of the <a href="https://github.com/kubernetes" rel="nofollow noopener">Kubernetes organization</a>. If I wanted to  suggest a change to something in the Kubernetes project, I would create a fork. <strong>But</strong> still I would create a branch locally in that fork, work on my changes, then push the branch to the GitHub-hosted repo, and open a pull request.</p>
<p>I hope that helps!</p>

**Comment by joshuaaguilar20** [comment]
<p>Thank you <a class="mention" href="/lee-dohm">@lee-dohm</a>. This is a beautiful explanation. Elegant yet simple.</p>
